cmake/InstallRules.cmake

```diff
@@ -0,0 +1,75 @@
+include(CMakePackageConfigHelpers)
+include(GNUInstallDirs)
+
+set(package ladybird)
+
+install(TARGETS LibAudio LibCore LibFileSystem LibGfx LibIPC LibJS LibWeb LibWebView LibWebSocket LibProtocol LibGUI LibMarkdown LibGemini LibHTTP LibGL LibSoftGPU LibVideo LibWasm LibXML LibIDL LibTextCodec LibCrypto LibLocale LibRegex LibSyntax LibUnicode LibCompress LibTLS LibGLSL LibGPU LibThreading LibSQL
+  EXPORT ladybirdTargets
+  LIBRARY
+    COMPONENT ladybird_Runtime
+    NAMELINK_COMPONENT ladybird_Development
+    DESTINATION ${CMAKE_INSTALL_LIBDIR}
+)
+
+install(TARGETS ladybird WebContent
+  EXPORT ladybirdTargets
+  RUNTIME
+    COMPONENT ladybird_Runtime
+    DESTINATION ${CMAKE_INSTALL_BINDIR}
+  LIBRARY
+    COMPONENT ladybird_Runtime
+    NAMELINK_COMPONENT ladybird_Development
+    DESTINATION ${CMAKE_INSTALL_LIBDIR}
+)
+
+write_basic_package_version_file(
+    "${package}ConfigVersion.cmake"
+    COMPATIBILITY SameMajorVersion
+)
+
+# Allow package maintainers to freely override the path for the configs
+set(
+  ladybird_INSTALL_CMAKEDIR "${CMAKE_INSTALL_DATADIR}/${package}"
+  CACHE PATH "CMake package config location relative to the install prefix"
+)
+mark_as_advanced(ladybird_INSTALL_CMAKEDIR)
+
+install(
+  FILES cmake/LadybirdInstallConfig.cmake
+  DESTINATION "${ladybird_INSTALL_CMAKEDIR}"
+  RENAME "${package}Config.cmake"
+  COMPONENT ladybird_Development
+)
+
+install(
+  FILES "${PROJECT_BINARY_DIR}/${package}ConfigVersion.cmake"
+  DESTINATION "${ladybird_INSTALL_CMAKEDIR}"
+  COMPONENT ladybird_Development
+)
+
+install(
+  EXPORT ladybirdTargets
+  NAMESPACE ladybird::
+  DESTINATION "${ladybird_INSTALL_CMAKEDIR}"
+  COMPONENT ladybird_Development
+)
+
+install(DIRECTORY
+    "${SERENITY_SOURCE_DIR}/Base/res/html"
+    "${SERENITY_SOURCE_DIR}/Base/res/fonts"
+    "${SERENITY_SOURCE_DIR}/Base/res/icons"
+    "${SERENITY_SOURCE_DIR}/Base/res/themes"
+    "${SERENITY_SOURCE_DIR}/Base/res/color-palettes"
+    "${SERENITY_SOURCE_DIR}/Base/res/cursor-themes"
+  DESTINATION "${CMAKE_INSTALL_DATADIR}/res"
+  USE_SOURCE_PERMISSIONS MESSAGE_NEVER
```

Comment:
Hm, so should we be installing to `res` actually? That's not a directory that exists on your convetional freedesktop system... `${CMAKE_INSTALL_DATADIR}/lagom` might be a better path (for what is `/res` in Serenity) perhaps?

Reply:
Yes, `/prefix/share/res` is what I meant. That is a path that makes sense for Serenity, but not for freedesktop systems.

Reply:
Same comment here, that's what we're doing upstream. If you have a better way to get the resources in an easily findable place then by all means patch it up. Slapping `s_serenity_resource_root` to the front of things was the easiest way to get it working.